core/src/main/java/org/apache/iceberg/avro/AvroFileAppender.java

```diff
@@ -78,14 +78,13 @@ public Metrics metrics() {
 
   @Override
   public long length() {
-    if (stream != null) {
-      try {
-        return stream.getPos();
-      } catch (IOException e) {
-        throw new RuntimeIOException(e, "Failed to get stream length");
-      }
+    Preconditions.checkNotNull(stream, "Failed to get stream length: no open stream");
+
+    try {
+      return stream.getPos();
+    } catch (IOException e) {
+      throw new RuntimeIOException(e, "Failed to get stream length");
```

Comment:
is this change actually related to this particular PR?

Reply:
Yes. It was also for a special case handling (RuntimeIO exception without IO exception cause).

Updated the PR title and description. So, it makes sense now.

Reply:
Sorry. I got confused. 

There was two `RuntimeIOException` from this length() method. Now, one is replaced with  `IllegalStateException` as per Dan's comment. So, there will still be one `RuntimeIOException` (which will be replaced with `UncheckedIOException` in the main PR as it has a cause of `IOException`)